### PR TITLE
Merge master into preview

### LINKED ISSUE #1132: [Preview] Twin desired property updates not received on AMQP
This issue affects the preview service API version. We are disabling these tests until service fixes this issue. ~Regular twin desired property updates are received successfully~

### LINKED ISSUE #1352: Devices.Client.SetDesiredPropertyUpdateCallbackAsync does not work using c# Device SDK 1.29.0-preview-004
- **OS, version, SKU and CPU architecture used:** <VERSION> Windows 10 Desktop x64
- **Application's .NET Target Framework :** <NETVERSION> netcoreapp3.1
- **Device:** <DEVICETYPE> Laptop
- **SDK version used:** <VERSION> 1.29.0-preview-004

## Description of the issue:
When using the latest preview c# device SDK the SetDesiredPropertyUpdateCallbackAsync callback does not fire when changing desired property using Azure IoT Central. When I role back to the latest stable release 1.25.0 the code does work.

## Code sample exhibiting the issue:
private const string ModuleConnectionString = "HostName=######.azure-devices.net;DeviceId=########;SharedAccessKey=###########";
        private static DeviceClient Client = null;

        static void Main(string[] args)
        {
            Microsoft.Azure.Devices.Client.TransportType transport =
              Microsoft.Azure.Devices.Client.TransportType.Amqp;

            try
            {
                Client = DeviceClient.CreateFromConnectionString(ModuleConnectionString, transport);
                Client.SetConnectionStatusChangesHandler(ConnectionStatusChangeHandler);
                Client.SetDesiredPropertyUpdateCallbackAsync(OnDesiredPropertyChanged, null).GetAwaiter().GetResult();

                Console.WriteLine("Retrieving twin");
                var twinTask = Client.GetTwinAsync();
                twinTask.Wait();
                var twin = twinTask.Result;
                Console.WriteLine(JsonConvert.SerializeObject(twin.Properties));

                Console.WriteLine("Sending app start time as reported property");
                TwinCollection reportedProperties = new TwinCollection();
                reportedProperties["DateTimeLastAppLaunch"] = DateTime.Now;

                Client.UpdateReportedPropertiesAsync(reportedProperties);
            }
            catch (AggregateException ex)
            {
                Console.WriteLine("Error in sample: {0}", ex);
            }

            Console.WriteLine("Waiting for Events.  Press enter to exit...");
            Console.ReadLine();
            Client.CloseAsync().Wait();
        }

        static void ConnectionStatusChangeHandler(ConnectionStatus status, ConnectionStatusChangeReason reason)
        {
            Console.WriteLine("Connection Status Changed to {0}; the reason is {1}",
              status, reason);
        }
        
        // This Task is never fired when using Device SDK 1.29.0-preview-004
        private static async Task OnDesiredPropertyChanged(TwinCollection desiredProperties, object userContext)
        {
            Console.WriteLine("desired property change:");
            Console.WriteLine(JsonConvert.SerializeObject(desiredProperties));
            Console.WriteLine("Sending current time as reported property");
            TwinCollection reportedProperties = new TwinCollection
            {
                ["DateTimeLastDesiredPropertyChangeReceived"] = DateTime.Now
            };

            await Client.UpdateReportedPropertiesAsync(reportedProperties).ConfigureAwait(false);
        }

## Console log of the issue:

AB#7281154

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Note this is a merge, not a rebase. The rebase was significantly harder, so I went with the merge instead. I'm pretty sure I did this last time, too

This merge removes twin array tests because the preview api version doesn't support them, and re-enables desired property over amqp tests since this merge brings in the fix from #1345 about the api version bump

Fixes #1132 
Fixes #1352

### CODE REVIEW

File: e2e/test/TwinE2ETests.cs [hunk 1]
```diff
@@ -92,16 +88,12 @@ public async Task Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEvent_WithObse
             await Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEventAsync(Client.TransportType.Mqtt_WebSocket_Only, SetTwinPropertyUpdateCallbackObsoleteHandlerAsync, Guid.NewGuid().ToString()).ConfigureAwait(false);
         }
 
-        // TODO: #1132 - Twin desired property updates not received after fault injection on AMQP
```
Comment (azabbasi) [inline e2e/test/TwinE2ETests.cs:94]: isn't this the same issue we noticed with the java SDK? that you will have to resend the AMQP messages when you open the links back up?

Comment (azabbasi) [reply e2e/test/TwinE2ETests.cs:94]: We should probably bring the fix to the C# SDK as well

Comment (timtay-microsoft) [reply e2e/test/TwinE2ETests.cs:94]: That is exactly what this PR does, actually. The fix for the C# SDK was checked into master, and now this merge brings that same fix to preview. That's why I took the time to re-enable any tests that were only disabled due to 1132

**Pull request merged**